JS.md

```diff
@@ -337,11 +341,34 @@ Among other things, this function performs the following observable steps:
   the final value is the last value written in order. Note: there should be no
   errors possible that would cause this operation to fail partway through. After
   this operation completes, elements of `instance` are visible and callable
-  through [imported Tables](Modules.md#imports), even if `start` fails.
+  through [imported tables](Modules.md#imports), even if `start` fails.
 
 * If a [`start`](Modules.md#module-start-function) is present, it is evaluated.
   Any errors thrown by `start` are propagated to the caller.
 
+The following steps are performed _before_ the `start` function executes:
+
+1. For each table 't' in [`instance.tables`](https://github.com/WebAssembly/spec/blob/master/interpreter/spec/instance.ml#L17):
+  1. If there is no element in `tables` whose `table.[[Table]]` is `t`:
+    1. Let `table` be a new `WebAssembly.Table` object with [[Table]] set to `t` and [[Values]] set to a new list of the same length all whose entries are `null`.
+    1. Append `table` to `tables`.
+  1. Otherwise:
+    1. Let `table` be the element in `tables` whose `table.[[Table]]` is `t`
+  1. (Note: At most one `WebAssembly.Table` object is created for any table, so the above `table` is unique, even if there are multiple occurrances in the list. Moreover, if the item was an import, the original object will be found.)
+  1. For each index `i` of `t`:
+    1. Let `c` be the `i`th element of `t`
+    1. If `c` is a [`closure`](https://github.com/WebAssembly/spec/blob/master/interpreter/spec/instance.ml#L7) `c`:
+      1. If there is an [Exported Function Exotic Object](#exported-function-exotic-objects) `in `funcs` whose `[[Closure]]` equals `c`:
+        1. Let `func` be that function object.
+      1. (Note: At most one wrapper is created for any closure, so `func` is uniquely determined. Moreover, if the item was an import that is already an [Exported Function Exotic Object](#exported-function-exotic-objects), then the original function object will be found. For imports that are regular JS functions, a new wrapper will be created.)
+      1. Otherwise:
+        1. Let `func` be an [Exported Function Exotic Object](#exported-function-exotic-objects) created from `c`.
+        1. Append `func` to `funcs`.
+        1. Return `func`.
```

Comment:
Since this isn't a reduction (and since the other branch doesn't have one), I'd remove the "Return" step.